### PR TITLE
Fixed 0/180 rotation coordinates

### PR DESCRIPTION
**Pull request by kattni**

### CODE REVIEW

**Comment by kattni** [comment]
Adds code to limit what NeoPixel coordinates are valid because some invalid `offsets` were coming up negative which was allowing it to light up an incorrect NeoPixel by counting backwards on the NeoPixel index.

Tested every NeoPixel coordinate on all four rotations, and tested every button press on all four rotations.

File: adafruit_trellism4.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def __init__(self, pin, *, width, height, rotation=0):
     def __setitem__(self, index, value):
         if not isinstance(index, tuple) or len(index) != 2:
             raise IndexError("Index must be tuple")
+        if index[0] > 7 or index[1] > 7:
```
Comment (dhalbert) [inline adafruit_trellism4.py]: I think could be a tighter check:
```
if index[0] >= self.width or index[1] >= self.height:
    raise IndexError(...)
```

`index[0]` is always the width, right?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks for the thorough testing!

File: adafruit_trellism4.py (hunk 1 above)
Comment (dhalbert) [reply adafruit_trellism4.py]: also check for the indexes < 0

Comment (kattni) [reply adafruit_trellism4.py]: Yes, and it looks like this works. Adding now!

**Review by dhalbert — APPROVED** [review]
Ok, great, and safer test for out of range at the end!

**Pull request merged**